Neos.Neos/Classes/Fusion/Cache/CacheTag.php

```diff
@@ -123,9 +133,11 @@ final public static function fromString(string $string): self
     }
 
     protected static function getHashForWorkspaceNameAndContentRepositoryId(
-        WorkspaceName $workspaceName,
+        WorkspaceName|CacheTagWorkspaceName $workspaceName,
         ContentRepositoryId $contentRepositoryId,
     ): string {
-        return sha1($workspaceName->value . '@' . $contentRepositoryId->value);
+        return sha1(
+            $workspaceName === CacheTagWorkspaceName::ANY ? $contentRepositoryId->value : $workspaceName->value . '@' . $contentRepositoryId->value
```

Comment:
```suggestion
            ($workspaceName === CacheTagWorkspaceName::ANY ? $contentRepositoryId->value : $workspaceName->value) . '@' . $contentRepositoryId->value
```

otherwise the precedence is wrong i think and the right side is concatenated to early

Reply:
This is actually intended:
true => contentRepositoryId
false => workspaceName@contentRepositoryId

Reply:
https://3v4l.org/J2nlQ

Reply:
👍